build.rs

```diff
@@ -13,6 +13,8 @@ mod asm {
     enum Arch {
         X86(ArchX86),
         Arm(ArchArm),
+        #[allow(dead_code)]
+        Unknown,
```

Comment:
Why do we need this?  Your `else { return; }` below should take care of it.

Reply:
Otherwise I get this: "warning: variant `Unknown` is never constructed".

Reply:
Sorry that might've been confusing.  I didn't mean the `#[allow(dead_code)]`.  Can we remove `Unknown`?  Then we can remove the `_ => unimplemented!("should not get here")`s that are never reached because of the `else { return; }`.